js/hw37.1.js

```diff
@@ -0,0 +1,76 @@
+const container = document.getElementsByClassName('flip-container')[0];
+const AnimDuration = 600;
+let cards = [];
+let animRunning = false;
+
+window.addEventListener('load', function () {
+    generateCards();
+    addCard();
+})
+
+container.addEventListener('mouseover', function (event) {
+    if (event.target.parentNode.parentNode.classList.contains('flipper') && !animRunning) {
+        addCard();
+        animRunning = true;
+        event.target.parentNode.parentNode.classList.toggle("level-up");
+        event.target.parentNode.parentNode.classList.add('animate');
+        sleep(AnimDuration).then(() => {
```

Comment:
What is `sleep` function used for doing and why use it?

Reply:
To prevent run new animation when animation for previous element not finished yet. It's unlikely situation but i was able to reproduce it.